contracts/core/OptionsPremiumPricer.sol

```diff
@@ -0,0 +1,146 @@
+//SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.7.3;
+
+import {IVolatilityOracle} from "../interfaces/IVolatilityOracle.sol";
+import {IPriceOracle} from "../interfaces/IPriceOracle.sol";
+import {ICompToken} from "../interfaces/ICompToken.sol";
+import {DSMath} from "../libraries/DSMath.sol";
+import {IBlackScholes} from "../interfaces/IBlackScholes.sol";
+import {IERC20Detailed} from "../interfaces/IERC20Detailed.sol";
+import {Math} from "../libraries/Math.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import "hardhat/console.sol";
+
+contract OptionsPremiumPricer is DSMath {
+    using SafeMath for uint256;
+
+    /**
+     * Immutables
+     */
+    IVolatilityOracle public immutable volatilityOracle;
+    IPriceOracle public immutable priceOracle;
+    ICompToken public immutable cToken;
+    // IKEEP3rVolatility: 0xCCdfCB72753CfD55C5afF5d98eA5f9C43be9659d
+    IBlackScholes public immutable blackScholes;
+
+    constructor(
+        address _volatilityOracle,
+        address _priceOracle,
+        address _cToken,
+        address _blackScholes
+    ) {
+        require(_volatilityOracle != address(0), "!_volatilityOracle");
+        require(_priceOracle != address(0), "!_priceOracle");
+        require(_cToken != address(0), "!_cToken");
+        require(_blackScholes != address(0), "!_blackScholes");
+        volatilityOracle = IVolatilityOracle(_volatilityOracle);
+        priceOracle = IPriceOracle(_priceOracle);
+        cToken = ICompToken(_cToken);
+        blackScholes = IBlackScholes(_blackScholes);
+    }
+
+    /**
+     * @notice Calculates the premium of the provided option using Black-Scholes
+     * References for Black-Scholes:
+       https://www.macroption.com/black-scholes-formula/
+       https://www.investopedia.com/terms/b/blackscholes.asp
+     * @param st is the strike price of the option
+     * @param expiryTimestamp is the unix timestamp of expiry
+     * @param isPut is whether the option is a put option
+     */
+    function getPremium(
+        uint256 st,
+        uint256 expiryTimestamp,
+        bool isPut
+    ) external view returns (uint256 premium) {
+        require(
+            expiryTimestamp > block.timestamp,
+            "Expiry must be in the future!"
+        );
+
+        uint256 sp = priceOracle.latestAnswer().div(10**priceOracle.decimals());
+        uint256 v = volatilityOracle.vol();
+        uint256 t = expiryTimestamp.sub(block.timestamp).div(1 days);
+
+        console.log("strike price is %s", st);
+        console.log("underlying price is %s", sp);
+        console.log("volatility is %s", v);
+        console.log("time to expiry in days is %s", t);
+
+        (uint256 call, uint256 put) = blackScholes.quoteAll(t, v, sp, st);
```

Comment:
some fundamental problems with blackScholes.quoteAll.

ex: if we input 
      a. t = 7 (it should be in days because [quoteAll](https://github.com/keep3r-network/keep3r.network/blob/a6897007db6e656e0e310ed7ee4ad42904fe2794/contracts/Keep3rV1Volatility.sol#L193) subsequently divides by 365.
      b. vol = 44478303 (disregard that this is wrong)
      c. sp (underlying) = 2000
      d. st (strike) = 2200

The call option costs more than the put option even though the call option would be OTM and the put would be ITM when purchased.

Other issues include call options with lower strike prices are **cheaper** than higher strike prices, expiries further out are cheaper. 

Bottom line not reliable, will have to change

Reply:
>The call option costs more than the put option even though the call option would be OTM and the put would be ITM when purchased.

>Other issues include call options with lower strike prices are cheaper than higher strike prices, expiries further out are cheaper.

Do we know the underlying reasons for why quoteAll is unreliable? They must have made mistakes in how they compute BS